### PR TITLE
add docs for byo certificate via domainmapping

### LINKED ISSUE #4081: Add documentation for BYO certificate for custom domains
**Describe the change you'd like to see**

- https://github.com/knative/serving/issues/10530 starts BYO certs but no documentation is available yet.
- We need a new doc page or add the procedure in https://knative.dev/docs/developer/serving/services/custom-domains/

### PR DESCRIPTION
**Pull request by shinigambit**
Fixes #4081

### CODE REVIEW

File: docs/developer/serving/services/byo-certificate.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Custom TLS certificate for DomainMapping
+
```
Comment (nak3) [inline docs/developer/serving/services/byo-certificate.md:2]: Should we add `{{ feature(beta="0.24") }}` like [Configuring custom domains section](https://github.com/knative/docs/blob/mkdocs/docs/developer/serving/services/custom-domains.md)?

File: docs/developer/serving/services/byo-certificate.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        name: <service-name>
+        kind: Service
+        apiVersion: serving.knative.dev/v1
+    # tls block specifies the secret to be used
+      tls:
+        secretName: <tls-secret-name>
+    ```
+    Where:
+
+    - `<tls-secret-name>` is the name of the TLS secret created in the previous step
```
Comment (nak3) [inline docs/developer/serving/services/byo-certificate.md:45]: nit
```suggestion
    - `<tls-secret-name>` is the name of the TLS secret created in the previous step.
```

File: docs/developer/serving/services/byo-certificate.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    The `URL` column of the status should show the mapped domain with the scheme updated to https:
+    ```
+    NAME                      URL                               READY   REASON
+    <domain-name>             https://<domain-name>             True
+    ```
+    1. Hit the service if exposed publicly:
+    ```bash
+    curl https://<domain-name>
+    ```
+    If the certificate is self-signed skip verification by adding the `-k` flag to curl
```
Comment (nak3) [inline docs/developer/serving/services/byo-certificate.md:65]: nit
```suggestion
    If the certificate is self-signed skip verification by adding the `-k` flag to curl.
```

**Review by nak3 — COMMENTED** [review]
LGTM

File: docs/developer/serving/services/byo-certificate.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# Custom TLS certificate for DomainMapping
+
+{{ feature(beta="0.24") }}
+
+By providing the reference to an existing _TLS Certificate_ you can instruct a `DomainMapping` to use that
+certificate to secure the mapped service, using this feature will skip _autoTLS_ certificate creation.
```
[the remark this answers is not in the document]
Comment (snneji) [reply docs/developer/serving/services/byo-certificate.md:6]: Also, do you think it would be nice to have a link to autoTLS here?

**Review by snneji — CHANGES_REQUESTED** [review]
Very nice PR, thanks @shinigambit! I just added a few small suggestions. 

One more thing, do you think that it would make sense to mention on the autoTLS page that you can use a custom certificate instead?

File: docs/developer/serving/services/byo-certificate.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+## Prerequisites
+
+- Same prerequisites as in [Configuring custom domains](../custom-domains) apply.
```
[the remark this answers is not in the document]
Comment (shinigambit) [reply docs/developer/serving/services/byo-certificate.md]: Thanks for your suggestions @snneji , I noticed that I put in the prerequisites an existing `TLS secret` but then in the procedure I put the steps to create one, so I simplified the requirement to just have a certificate. Let me know if it looks right

**Review by snneji — APPROVED** [review]

**Pull request merged**